statsmodels/tsa/stl/mstl.py

```diff
@@ -0,0 +1,245 @@
+"""
+Author: Kishan Manani
+License: BSD-3 Clause
+
+An implementation of MSTL [1], an algorithm for time series decomposition when
+there are multiple seasonal components.
+
+This implementation has the following differences with the original algorithm:
+- Missing data must be handled outside of this class.
+- The algorithm proposed in the paper handles a case when there is no
+seasonality. This implementation assumes that there is at least one seasonal
+component.
+
+[1] K. Bandura, R.J. Hyndman, and C. Bergmeir (2021)
+MSTL: A Seasonal-Trend Decomposition Algorithm for Time Series with Multiple
+Seasonal Patterns
+https://arxiv.org/pdf/2107.13462.pdf
+"""
+from typing import Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from scipy.stats import boxcox
+
+from statsmodels.tools.typing import ArrayLike1D
+from statsmodels.tsa.seasonal import STL
```

Comment:
You should directly import STL from where it is defined, not from `seasonal` which is just an API end-point. This can produce cyclic import problems.